InfiniTAM/Engine/PoseSourceEngine.cpp

```diff
@@ -2,57 +2,42 @@
 
 #include "PoseSourceEngine.h"
 
-#include "../Utils/FileUtils.h"
-
+#include <glog/logging.h>
 #include <stdio.h>
 
-using namespace InfiniTAM::Engine;
+#include "../Utils/FileUtils.h"
 
-PoseSourceEngine::PoseSourceEngine(const char* poseMask) {
-  strncpy(this->poseMask, poseMask, BUF_SIZE);
 
+namespace InfiniTAM {
+namespace Engine {
+PoseSourceEngine::PoseSourceEngine() {
   currentFrameNo = 0;
   cachedFrameNo = -1;
 
   cached_pose = NULL;
+  LOG(INFO) << "PoseSourceEngine \n";
+
 }
 
 void PoseSourceEngine::loadPoseIntoCache(void) {
-  char str[2048];
-  FILE* f;
-  bool success = false;
-
-  cached_pose = new ITMPose();
-
-  sprintf(str, poseMask, currentFrameNo);
-  f = fopen(str, "r");
-  if (f) {
-    size_t ret =
-        fscanf(f, "%f %f %f %f %f %f %f", /*TODO(gocarlos): Implement this*/);
-
-    fclose(f);
-
-    if (ret == 9) success = true;
-  }
-
-  if (!success) {
-    delete cached_pose;
-    cached_pose = NULL;
-    printf("error reading file '%s'\n", str);
-  }
+  // TODO(gocarlos): Implement this
```

Comment:
Yes, if you would do that you probably wouldn't need the hack of having a pointer to the main engine, I guess, but it's probably fine for now. (If you're going to use the GPU you probably have to rethink this)